portality/models/harvester.py

```diff
@@ -0,0 +1,243 @@
+from builtins import object
+from portality.lib import dataobj
+from portality.dao import DomainObject
+from portality.lib import dates, es_data_mapping
+from portality.core import app
+
+
+class HarvesterPlugin(object):
+    def get_name(self):
+        raise NotImplementedError()
+
+    def iterate(self, issn, since, to=None):
+        """
+        Iterate over the records associated with the issn from "since" until "to"
+
+        This should return a generator (i.e. it should yield), and it needs to yeild a tuple
+        containing (<doaj article>, <harvest date for this record>)
+
+        :param issn:
+        :param since:
+        :param to:
+        :return:
+        """
+        raise NotImplementedError()
+
+
+class HarvestState(dataobj.DataObj, DomainObject):
+    __type__ = 'harvester_state'
+
+    def __init__(self, **raw):
+        struct = {
```

Comment:
so far by convention we've defined our structs as constants rather than in __init__ - I guess it doesn't matter really, apart from that it's easier to find if it's not simply called `struct`

Reply:
seems sensible to be consistent in the main codebase